version: '3.4'
services:

  fastapi:
    image: fastapi
    build:
      context: .
      dockerfile: ./Dockerfile
    # volumes:
    #   - ./src/bank_of_tomorrow:/app
    command: ["sh", "-c", "python -m uvicorn main:app --host 0.0.0.0 --port 8000 --reload"]
    environment:
      - POSTGRES_URI=postgresql://postgres:postgres@desarrollo-postgres-1:5432/postgres
    ports:
      - 8000:8000
    depends_on:
      postgres:
        condition: service_healthy

  postgres:
    image: postgres
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=postgres
    volumes:
      - ./db_config/:/docker-entrypoint-initdb.d/
    ports:
      - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready --dbname $$POSTGRES_DB --username $$POSTGRES_USER"]
      interval: 10s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4
    environment:
      - PGADMIN_DEFAULT_EMAIL=tomas@gmail.com
      - PGADMIN_DEFAULT_PASSWORD=123
    ports:
      - "8080:80"
    depends_on:
      - postgres